### PR TITLE
feat(bridge-ui): Recommend fee

### PR DESCRIPTION
**Pull request by cyberhorsey**

### CODE REVIEW

**Review by dantaik — DISMISSED** [review]

File: packages/bridge-ui/src/eth/bridge.spec.ts [hunk 1]
```diff
@@ -160,6 +161,8 @@ describe("bridge tests", () => {
         message: {
           owner: "0x",
           srcChainId: BigNumber.from(167001),
+          destChainId: BigNumber.from(31336),
```
Comment (d1onys1us) [inline packages/bridge-ui/src/eth/bridge.spec.ts:164]: we should probably use a constant here now.

File: packages/bridge-ui/src/utils/recommendProcessingFee.spec.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
+const mockChainIdToTokenVaultAddress = jest.fn();
+jest.mock("../store/bridge", () => ({
```
Comment (d1onys1us) [inline packages/bridge-ui/src/utils/recommendProcessingFee.spec.ts:2]: should this be below the imports? stylistically

File: packages/bridge-ui/src/utils/recommendProcessingFee.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      TokenVault,
+      signer
+    );
+
+    const bridged = await tokenVault.canonicalToBridged(
+      toChain.id,
+      srcChainAddr
+    );
+
+    // gas limit for erc20 if not deployed on the dest chain already is about ~2.9m
```
Comment (d1onys1us) [inline packages/bridge-ui/src/utils/recommendProcessingFee.ts:44]: nit: abide by 80 column width comments

File: packages/bridge-ui/src/utils/recommendProcessingFee.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  fromChain: Chain,
+  feeType: ProcessingFeeMethod,
+  token: Token,
+  signer: Signer
+): Promise<string> {
+  if (!toChain || !fromChain || !token || !signer || !feeType) return "0";
+  const p = get(providers);
+  const provider = p.get(toChain.id);
+  const gasPrice = await provider.getGasPrice();
+  let gasLimit = ethGasLimit; // gasLimit for processMessage call for ETH is about ~800k. to make it enticing, we say 900k.
```
Comment (d1onys1us) [inline packages/bridge-ui/src/utils/recommendProcessingFee.ts:26]: nit: abide by 80 column width comments

**Review by dionysuzx — DISMISSED** [review]

File: packages/bridge-ui/src/utils/recommendProcessingFee.spec.ts (hunk 2 above)
Comment (cyberhorsey) [reply packages/bridge-ui/src/utils/recommendProcessingFee.spec.ts:2]: no, this mock of the partial page actually has to be above imports

File: packages/bridge-ui/src/eth/bridge.spec.ts (hunk 1 above)
Comment (cyberhorsey) [reply packages/bridge-ui/src/eth/bridge.spec.ts:164]: this is just a test file

Comment (d1onys1us) [reply packages/bridge-ui/src/eth/bridge.spec.ts:164]: i still think that TAIKO_CHAIN_ID etc. is slightly more semantic, but yeah not a big deal anyways.

**Review by dionysuzx — APPROVED** [review]

**Review by dantaik — APPROVED** [review]

**Pull request merged**